fcm/fcm.go

```diff
@@ -98,56 +97,45 @@ func (f *FCM) ConnectToFcm(fcmNotifications chan<- notification.PrinterNotificat
 		reader := bufio.NewReader(resp.Body)
 		go func() {
 			for {
-				raw_input, err1 := reader.ReadBytes('\n')
-				input_chunks := strings.SplitN(string(raw_input), "\n", 2)
-				notification_size := strings.TrimSpace(input_chunks[0])
-				notification_data1 := ""
-				if len(input_chunks) > 1 {
-					notification_data1 = strings.TrimSpace(input_chunks[1])
+				printerId, err := GetPrinterID(reader)
+				if len(printerId) > 0 {
+					pn := notification.PrinterNotification{printerId, notification.PrinterNewJobs}
+					fcmNotifications <- pn
 				}
-				size, _ := strconv.Atoi(notification_size)
-				buffer_size := size - len(notification_data1)
-
-				notification_data2 := ""
-				var err2 error
-				for 0 != buffer_size {
-					// part of notification
-					notification_data2_buffer := make([]byte, buffer_size)
-					n, err2 := reader.Read(notification_data2_buffer)
-					notification_data2 += string(notification_data2_buffer)
-					buffer_size -= n
-
-					if err2 != nil {
-						break
-					}
-				}
-				// process EOF signal after processing notification.
-				if err1 == io.EOF || err2 == io.EOF || (err2 == nil && err1 == nil) {
-					notification_string := notification_data1 + notification_data2
-					if len(notification_string) > 0 {
-						printerId := GetPrinterID(notification_string)
-						if printerId != "" {
-							pn := notification.PrinterNotification{printerId, notification.PrinterNewJobs}
-							fcmNotifications <- pn
-						}
-					}
-					if err2 == io.EOF || err1 == io.EOF {
+				if err != nil {
 						log.Info("DRAIN message received, client reconnecting.")
 						dead <- struct{}{}
 						break
 					}
-				} else {
-					// stop listening unknown error happened.
-					log.Errorf("Unexpected error happened on FCM listener: %v, %v", err1, err2)
-					quit <- struct{}{}
-					break
-				}
 			}
 		}()
 	}
 	return nil
 }
 
+func GetPrinterID(reader *bufio.Reader) (string, error) {
+	raw_input, err := reader.ReadBytes('\n')
+	if err == nil {
+		buffer_size, _ := strconv.Atoi(strings.TrimSpace(string(raw_input)))
```

Comment:
Rather than call TrimSpace, can you just slice raw_input to remove the final char (which is guaranteed to be '\n')?

Reply:
Done.